sql/plan/ddl.go

```diff
@@ -81,10 +96,29 @@ func (c *CreateTable) RowIter(ctx *sql.Context) (sql.RowIter, error) {
 	creatable, ok := c.db.(sql.TableCreator)
 	if ok {
 		err := creatable.CreateTable(ctx, c.name, c.schema)
-		if sql.ErrTableAlreadyExists.Is(err) && c.ifNotExists {
-			err = nil
+		if err != nil && !(sql.ErrTableAlreadyExists.Is(err) && c.ifNotExists) {
+			return sql.RowsToRowIter(), err
+		}
+		if len(c.fkDefs) > 0 {
```

Comment:
In light of our other conversation, add a TODO here noting the fact that this will create a table without keys in the case that an integrator doesn't support foreign keys on their tables, and that we need transaction / rollback boundaries (or something else) to prevent this.